nodes/quantum/classical-register/classical-register.js

```diff
@@ -89,22 +76,31 @@ module.exports = function(RED) {
             circuitScript = util.format(circuitScript, register.registerVar);
           });
 
-          // Run the script in the python shell, and if no error occurs
-          // then notify the runtime when the node is done.
-          await shell.execute(circuitScript, (err) => {
-            logger.trace(node.id, 'Executed classical register command');
-            if (err) {
-              logger.error(node.id, err);
-              done(err);
-            } else {
-              done();
-            };
-          });
+          script += circuitScript;
         }
       }
 
+      // Run the script in the python shell, and if no error occurs
+      // then notify the runtime when the node is done.
+      await shell.execute(script, (err) => {
+        logger.trace(node.id, 'Executed classical register command');
+        if (err) {
+          error = err;
+        } else {
+          error = null;
+        }
+      });
+
+      if (error) {
+        logger.error(node.id, error);
+        done(error);
+        return;
+      }
+
       // update quantum circuit config
       quantumCircuitConfig[node.name] = register;
+      send(output);
+      done();
```

Comment:
Could probably simplify this to:
```suggestion
      // Run the script in the python shell, and if no error occurs
      // then notify the runtime when the node is done.
      await shell.execute(script, (err) => {
        logger.trace(node.id, 'Executed classical register command');
        if (err) {
          logger.error(node.id, err);
          done(err);
        } else {
          quantumCircuitConfig[node.name] = register;
          send(output);
          done();
        }
      });
```